src/interfaces.ts

```diff
@@ -88,10 +96,22 @@ export interface DefiSDKResponse {
   errors: string[] | undefined;
 }
 
+export interface UnderlyingTokenBalance {
+  token: UnderlyingToken;
+  amount: number;
+  price: number;
+  value: number;
+}
+
 export interface TokenBalance {
   token: Token;
   amount: number;
   price: number;
   value: number;
   percentOfTotal?: number;
 }
+
+export interface Protocol {
```

Comment:
There should be ProtocolTokenBalance as 
```ts
interface ProtocolTokenBalance {
  token: Token (protocol token you get from Covalent (Ex: UNI-V2 USDC-DAI))
  underlyingTokenBalances: UnderlyingTokenBalance[]
}
```

And we can have Protocol with 
```ts
interface Protocol {
type: ProtocolType (enum with type Uniswap, Yearn, Aave, etc)
protocolTokenBalances: ProtocolTokenBalance[]!
}
``` 

We can then maintain array of `Protocol` in a state and use it in protocol components